internal/provider/resource_locally_signed_cert_test.go

```diff
@@ -297,7 +297,143 @@ func TestAccLocallySignedCertNotRecreatedForEarlyRenewalUpdateInFuture(t *testin
+						return fmt.Errorf("incorrect IP address 0: expected %v, got %v", expected, got)
+					}
+					if expected, got := 2, len(cert.URIs); got != expected {
+						return fmt.Errorf("incorrect number of URIs: expected %v, got %v", expected, got)
+					}
+					if expected, got := "spiffe://example-trust-domain/workload", cert.URIs[0].String(); got != expected {
+						return fmt.Errorf("incorrect URI 0: expected %v, got %v", expected, got)
+					}
+					if expected, got := "spiffe://example-trust-domain/workload2", cert.URIs[1].String(); got != expected {
+						return fmt.Errorf("incorrect URI 1: expected %v, got %v", expected, got)
+					}
+
+					if expected, got := 2, len(cert.ExtKeyUsage); got != expected {
+						return fmt.Errorf("incorrect number of ExtKeyUsage: expected %v, got %v", expected, got)
+					}
+					if expected, got := x509.ExtKeyUsageServerAuth, cert.ExtKeyUsage[0]; got != expected {
+						return fmt.Errorf("incorrect ExtKeyUsage[0]: expected %v, got %v", expected, got)
+					}
+					if expected, got := x509.ExtKeyUsageClientAuth, cert.ExtKeyUsage[1]; got != expected {
+						return fmt.Errorf("incorrect ExtKeyUsage[1]: expected %v, got %v", expected, got)
+					}
+
+					if expected, got := x509.KeyUsageKeyEncipherment|x509.KeyUsageDigitalSignature, cert.KeyUsage; got != expected {
+						return fmt.Errorf("incorrect KeyUsage: expected %v, got %v", expected, got)
+					}
+
+					// This time checking is a bit sloppy to avoid inconsistent test results
+					// depending on the power of the machine running the tests.
+					now := time.Now()
+					if cert.NotBefore.After(now) {
+						return fmt.Errorf("certificate validity begins in the future")
+					}
+					if now.Sub(cert.NotBefore) > (2 * time.Minute) {
+						return fmt.Errorf("certificate validity begins more than two minutes in the past")
+					}
+					if cert.NotAfter.Sub(cert.NotBefore) != time.Hour {
+						return fmt.Errorf("certificate validity is not one hour")
+					}
+
+					caBlock, _ := pem.Decode([]byte(testCACert))
+					caCert, err := x509.ParseCertificate(caBlock.Bytes)
+					if err != nil {
+						return fmt.Errorf("error parsing ca cert: %s", err)
+					}
+					certPool := x509.NewCertPool()
+
+					// Verify certificate
+					_, err = cert.Verify(x509.VerifyOptions{Roots: certPool})
+					if err == nil {
+						return errors.New("incorrectly verified certificate")
+					} else if _, ok := err.(x509.UnknownAuthorityError); !ok {
+						return fmt.Errorf("incorrect verify error: expected UnknownAuthorityError, got %v", err)
+					}
+					certPool.AddCert(caCert)
+					if _, err = cert.Verify(x509.VerifyOptions{Roots: certPool}); err != nil {
+						return fmt.Errorf("verify failed: %s", err)
+					}
+
+					return nil
+				},
+			},
+		},
+	})
+}
+
+func locallySignedCertConfig(validity, earlyRenewal uint32, setKeyAlgorithm bool) string {
+	caKeyAlgorithmCfg := ""
+	if setKeyAlgorithm {
+		caKeyAlgorithmCfg = `ca_key_algorithm = "RSA"`
+	}
```

Comment:
Rather than implementing this logic in Go it could also be done in the configuration itself with conditional logic, e.g.

```
ca_key_algorithm = %t ? "RSA" : null
```

Although admittedly, most providers prefer to create wholly separate configurations for clarity following the Go practice of preferring some duplication for clarity over being purely DRY. (It makes much more sense when you're working with dozens or hundreds of tests so you don't need to figure out any special testing configuration logic.)

Reply:
Given those tests are only temporarily there, is it ok if I punt on it? I intend to release 4.x soon after this.

Reply:
Totally! Not a big deal, just wanted to let you know. 😄